### DISCUSSION TITLE
Extend the library to Android/iOs

### DISCUSSION THREAD

**Discussion by TopperDEL** [ffmpeginteropx/FFmpegInteropX#229] [Ideas]
Would it be feasable to also create the binaries for iOs and Android? There currently is no single library to rule them all - but I also do not really understand how your binding work on the low-level-side and if it would be possible to extend it to other platforms.

**Comment by brabebhin** [comment]
Hi.

Unless Microsoft makes it so uwp can cross compile for the other platforms, it would take quite a bit of time to support android and ios.

For Android there's exoplayer that supports a ffmpeg extension. Not sure about ios.

**Comment by TopperDEL** [comment]
I would like to use you library with the Uno Platform, which makes it that UWP gets cross compiled. All I would need are binaries for Android and iOs.
I'll have a look at exoplayer, though. Basically I just need the FrameGrabber and that seems to not always be part of the other FFMPEG-libraries out there. And it seems only one other is available for iOs.

**Comment by lukasf** [comment]
The thing is that Uno Platform only works with C#/.NET, while our lib is a native implementation against native WinRT interfaces. I don't believe that there are any plans in any project to provide native UWP interfaces for other platforms.

Uno has support for LibVLCSharp though. It seems to support frame grabbing:

https://code.videolan.org/mfkl/libvlcsharp-samples/tree/master/PreviewThumbnailExtractor

**Comment by TopperDEL** [comment]
Ah, I wasn't aware that you library relies on WinRT interfaces. From my understand you're more or eless embedding the c-files - but I didn't really get it.

LibVLC should do it - but I could not make it to work on UWP due to strange errors. iOs and Android crashed, too, and I did not yet find the time to analyze it in depth. I switched over to platform-specific implementations using FFMpeg on UWP (your one), MediaMetadataRetriever on Android and AVAssetImageGenerator on iOs.

**Comment by lukasf** [comment]
If I were to develop a cross-platform media app, I would probably use the following approach: Have a high-level abstraction in the app plus a low-level implementation that uses the best media pipeline available in each platform. This is probably what you are doing now as well. It is difficult already to create a media pipeline that works well for a specific platform. Creating one that works flawless and with great performance on every platform is a very difficult task.

**Reply by TopperDEL** [reply]
That's what I just painfully found out. :)

Thanks for your help and insights! At least for UWP the task I wanted to complete was flawlessly with your lib! Thank you!
